sdk/inc/azure/core/az_credentials.h

```diff
@@ -125,6 +127,20 @@ AZ_NODISCARD az_result az_credential_client_secret_init(
     az_span client_id,
     az_span client_secret);
 
+/**
+ * @brief sets authentication authority URL for the client secret credential.
+ *
+ * @param ref_credential reference to an az_credential_client_secret instance to set the authority
+ * URL for
+ * @param authority authentication authority URL to set
+ *
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ *         - Other error code if initialization failed
+ */
+AZ_NODISCARD az_result
+az_credential_set_authority(az_credential_client_secret* ref_credential, az_span authority);
```

Comment:
It could be useful to the caller to know from the docs when to call this and what possible values to pass in. For example, maybe explain and add a link to the national cloud doc.

Reply:
Updated.